centralstore/redis_store.go

```diff
@@ -1002,6 +1027,8 @@ func newTraceStateChangeEvent(current, next CentralTraceState) stateChangeEvent
 	}
 }
 
+// string returns a string representation of the state change event
+// this formatting logic should match the one in the stateChange lua script
```

Comment:
I know that Go will be faster with `return s.current + "-" + s.next`. I'm reasonably sure that in Lua, it will be faster to say `local stateChangeEvent = currentState .. "-" .. nextState`.

You should also echo this comment in the lua code that it should be the same as the go code.